xc7/utils/prjxray_arch_import.py

```diff
@@ -166,27 +182,37 @@ def main():
         with open(args.synth_tiles) as f:
             synth_tiles = json.load(f)
 
+        # Map ROI coordinates to the target VPR grid
+        roi_loc_lo = grid_loc_mapper.get_vpr_loc((j['info']['GRID_X_MIN'], j['info']['GRID_Y_MIN']))
+        roi_loc_hi = grid_loc_mapper.get_vpr_loc((j['info']['GRID_X_MAX'], j['info']['GRID_Y_MAX']))
+
         roi = Roi(
-                db=db,
-                x1=j['info']['GRID_X_MIN'],
-                y1=j['info']['GRID_Y_MIN'],
-                x2=j['info']['GRID_X_MAX'],
-                y2=j['info']['GRID_Y_MAX'],
-                )
+            db=db,
+            x1=min([p[0] for p in roi_loc_lo]),  # One physical grid location may map to more than one
+            y1=min([p[1] for p in roi_loc_lo]),  # VPR locations. So here we take min and max.
+            x2=max([p[0] for p in roi_loc_hi]),
+            y2=max([p[1] for p in roi_loc_hi]),
+        )
 
         synth_tile_map = add_synthetic_tile(complexblocklist_xml)
 
     for loc in g.tile_locations():
         gridinfo = g.gridinfo_at_loc(loc)
         tile = g.tilename_at_loc(loc)
 
+        # Map tile location to the VPR grid
+        loc = g.loc_of_tilename(tile)
+        vpr_loc = grid_loc_mapper.get_vpr_loc((loc.grid_x, loc.grid_y))
+        vpr_loc = vpr_loc[0] # FIXME: Assuming no split of that tile!
+        vpr_loc = GridLoc(vpr_loc[0], vpr_loc[1])
+
         if tile in synth_tiles['tiles']:
             synth_tile = synth_tiles['tiles'][tile]
 
             assert len(synth_tile['pins']) == 1
 
             vpr_tile_type = synth_tile_map[synth_tile['pins'][0]['port_type']]
-        elif only_emit_roi and not roi.tile_in_roi(loc):
+        elif only_emit_roi and not roi.tile_in_roi(vpr_loc):
```

Comment:
I think this is a case where we want to map from VPR location to physical location, and leave the ROI definition alone?  Thoughts?

Reply:
I already changed that in `prjxray_create_synth_tiles.py`. I will do the same for this script.